docassemble/ALKilnInThePlayground/data/questions/run_alkiln_tests.yml

```
@@ -2159,83 +2552,428 @@ content: |
+      _debug(f"""🖊️ ALKiln NOTE: { message }""")
+      return {
+        "ok": False, "message": message, "result": result
+      }
+
+        # try:
+        #   result = {}
+        #   json.dump(result, config_file, ensure_ascii=False, indent=2)
+  
+        # except Exception as config_dump_error:
+        #   message = f"""Error dumping json to config file at "{ local_config_path }":\n{ "\n".join(traceback.format_exception( config_dump_error )) }"""
+        #   _log(f"""🤕 ALKiln ERROR ALKP0015: { message }""")
+        #   return {
+        #     "ok": False, "message": message, "result": result
+        #   }
+  
+    return { "ok": True, "message": message, "result": result }
+---
+reconsider: True
+need:
+  - get_config_path
+  # - get_existing_path
+code: |
+  """
+  Return the runtime_config.json path or, if it doesn't yet exist, None
+  """
+  # temp_config_path = get_existing_path( "runtime_config.json" )
+  # if temp_config_path is None:
+  #   config_path_problem = 'No "runtime_config.json" file exists.'
+  #   _log(f'🤕 ALKiln ERROR: { config_path_problem }')
+  # config_path = temp_config_path
+  config_path = get_config_path()["result"]
 ---
 need:
   - get_existing_path
 code: |
-  temp_config_path = get_existing_path( "runtime_config.json" )
-  if temp_config_path is None:
-    config_path_problem = 'No "runtime_config.json" file exists.'
-    _log(f'🤕 ALKiln ERROR: { config_path_problem }')
-  config_path = temp_config_path
+  def get_config_path():
+    """
+    Return the runtime_config.json path or, if it doesn't yet exist, None
+    """
+    ok = True
+    message = ""
+    result = get_existing_path( "runtime_config.json" )
+  
+    if result is None:
+      ok = False
+      message = 'No "runtime_config.json" file exists.'
+      _log(f'🔎 ALKiln WARNING: { message }')
+    return { "ok": ok, "message": message, "result": result }
 ---
 id: default config_path_problem
 code: |
-  config_path_problem = ''
+  config_path_problem = ''  # Default
 ---
 # We get the folder_path by using ALKiln's runtime_config.json
 # folder_path tries different locations till it finds the real one in order to stay compatibile with old ALKiln versions. See `get_existing_path()` notes. We use `folder_path` to get the artifacts folder to make zip names, zips, and to eventually delete the folder.
-id: artifacts folder name
+id: artifacts paths and names
 need:
   - config_path
   - get_existing_path
 code: |
   import os
+  import traceback
+
+  # _debug(f"""folder_name trace: {"\n".join( traceback.print_stack() )}""")
   
   if config_path is None:
     folder_temp_name = 'no config'
 
   else:
-    with open( config_path ) as config_text:
+    with open( config_path ) as config_file:
 
       try:
-        folder_temp_name = json.load( config_text ).get('artifacts_path', 'no value')
+        folder_temp_name = json.load( config_file ).get('artifacts_path', 'no json value')
```

Maybe we should just `raise` here instead. Here's some language regardless.

```suggestion
        folder_temp_name = json.load( config_file ).get('artifacts_path', 'no json value')
        folder_path_problem = f"""Unable to get test results files. The ALKiln runtime_config.json file at "{ config_path }" has no "artifacts_path" value."""
```